NW | ITP_May-2025 | Sangeetha Devi Cuddapha Desaiah | Module-Onboarding | Form-Controls

sangeethadevi-cuddaphadesaiah opened the pull request:
<!--

You must title your PR like this:

REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME

For example,

London | May-2025 | Carol Owen | Wireframe

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [ ] I have committed my files one by one, on purpose, and for a reason
- [ ] I have titled my PR with REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME
- [ ] I have tested my changes
- [ ] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
- [ ] My changes meet the [requirements](./README.md) of this task

## Changelist

Designed form page based on acceptance criteria.

## Questions

Ask any questions you have for your reviewer.

cjyuan commented:
A well-prepared PR makes it easier for reviewers to approve it with minimal back-and-forth.
The "PR Essentials" file on the [Canvas](https://codeyourfutur-yov6609.slack.com/archives/C07QX99JK7B) in the #cyf-code-review channel has some helpful tips to make your PRs more robust and ready for review.
Can you take a few minutes to check against the essentials, including how to prepare PR description? Doing so can help speed up the review process.

And as a practice to get better at using AI tools, try running your code by ChatGPT. You can ask ChatGPT
```
review HTML code:
...        <--- paste your HTML code here
```

Then take a look at the suggestions it gives you, and pick out the ones that actually make sense for your project. It's a great way to learn and improve your work.

Once you've finished making changes, please leave a comment on this PR so I’ll get notified. Thanks!

sangeethadevi-cuddaphadesaiah commented:
Hi @cjyuan , As per your comments I have implemented changes and pushed the code. Please review it.
-Validate html code 
-Handled code quality
-Used AI for pre-review checks
-Changed PR description
-Review modified files thoroughly
-Apply the “Needs Review” label

cjyuan commented on Form-Controls/index.html:
```diff
@@ -1,27 +1,77 @@
+            <strong><span aria-label="required">*</span></strong>
+          </label> -->
+        <label for="name">Customer Name: <span aria-hidden="true">*</span></label>
+        <input type="text" id="name" name="name" required minlength="2" maxlength="50" pattern="[A-Za-z\s]+"
+          placeholder="Enter your name"
+          oninvalid="setCustomValidity('Name must be 2 characters and contain only letters.')"
+          oninput="setCustomValidity('')">
```
Interesting feature but the spec does say no JavaScript though.

You don't have to change this, but in practice I would recommend better to to fully adhere to the specification, or at minimum, seek prior approval before introducing features you believe are beneficial but may risk breaking the spec.

cjyuan commented on Form-Controls/index.html:
```diff
@@ -1,27 +1,77 @@
+            <strong><span aria-label="required">*</span></strong>
+          </label> -->
+        <label for="name">Customer Name: <span aria-hidden="true">*</span></label>
+        <input type="text" id="name" name="name" required minlength="2" maxlength="50" pattern="[A-Za-z\s]+"
```
With this pattern, a user can enter a name containing only space characters.

Can you try using ChatGPT to suggest a pattern (which is a regular expression) that match your constraint,
`Name must be 2 characters and contain only letters` ?

sangeethadevi-cuddaphadesaiah replied:
I have handled the name validation using  stricter regex pattern and pushed the code. Could you please review it.

Thanks.

cjyuan reviewed (commented):
Code is error free and well indented. Great job!

Form is well prepared! Can you enforce a stricter naming rule by updating the pattern?

cjyuan commented:
Changes look good! I hope you know what that pattern really means. :)

sangeethadevi-cuddaphadesaiah commented:
Yes , Now I can understand how to write the pattern with set of rules in html code.
Thanks @cjyuan .
